src/vpic/dump.cc

```diff
@@ -64,6 +71,13 @@ void vpic_simulation::predicate_copy(species_t* sp_from, species_t* sp_to, std::
                 if(sp_from->has_ids && sp_to->has_ids) {
                   sp_to->p_id[next] = sp_from->p_id[i];
                 }
+                #ifdef VPIC_PARTICLE_ANNOTATION
+                if(sp_from->has_annotation && sp_to->has_annotation) {
+                  for(int a = 0; a < sp_from->has_annotation && a < sp_to->has_annotation; a++) {
```

Comment:
Do we ever expect `sp_from->has_annotation` and `sp_to->has_annotation` to be different? 

Also do they default to -1 op 0? (if 0 we can remove the if?)

Reply:
I am scared that a user will call `init_buffered_particle_dump` before they call `sp->allocate_annotation_buffer`.